torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,881 @@ namespace fuser {
+                            &split](decltype(find_outer_it) it) {
+        if (it != ids.begin()) {
+          ids.erase(ids.begin(), it);
+        }
+
+        // Set outer it and position
+        find_outer_it = std::find(ids.begin(), ids.end(), split->outer());
+        outer_pos = find_outer_it == ids.end()
+            ? ids.size()
+            : std::distance(ids.begin(), find_outer_it);
+
+        // Set inner it and position
+        find_inner_it = std::find(ids.begin(), ids.end(), split->inner());
+        inner_pos = find_inner_it == ids.end()
+            ? ids.size()
+            : std::distance(ids.begin(), find_inner_it);
+      };
+
+      // Dry run to fill state
+      clear_left_of(ids.begin());
+
+      // Check if the domains out of the split are contiguous in the mapped
+      // domain.
+      if (find_outer_it == ids.end() && find_inner_it != ids.end()) {
+        // Outer dimension was not found, but inner dimension was. Must assume
+        // everything to the left of inner is not contiguously merged.
+        //
+        // Clear left of inner
+        clear_left_of(find_inner_it);
+      } else if (find_outer_it != ids.end() && find_inner_it == ids.end()) {
+        // Inner dimension was not found, outer and anything left of outer are
+        // definitely not contiguous.
+        //
+        // Clear outer and left of outer
+        clear_left_of(find_outer_it + 1);
+        continue;
+      } else if (find_outer_it == ids.end() && find_inner_it == ids.end()) {
+        // Nothing mapped, just continue
+        continue;
+      }
+
+      if (find_outer_it != ids.end() && find_inner_it != ids.end()) {
+        // Both outer and inner mapped.
+        if (outer_pos >= inner_pos) {
+          // Make sure outer is outside inner, otherwise neither could be part
+          // of a continuous mapping. There are cases where we could have
+          // reversible operations e.g.:
+          //    [id{3} id{5} id{6}] -> merge(1, 0)
+          // -> [id{5*3} id{6}] -> split(0, 5)
+          // -> [id{5} id{3} id{6}] -> transpose(0, 1)
+          // -> [id{3} id{5} id{6}]
+          // However we don't try and capture cases like this correcly, we'd
+          // just reduce this down to only the iter domain of size 6 mapping.
+          //
+          // Clear outer and left of outer
+          clear_left_of(find_outer_it + 1);
+          continue;
+        }
+      }
+
+      // Find the position inner would have to have to be considered ordered
+      // relative to outer
+      auto pos_after_outer = outer_pos + 1;
+      for (; pos_after_outer < ids.size(); pos_after_outer++) {
+        if (ids[pos_after_outer]->isBroadcast()) {
+          // Skip broadcast axes as they must not have been concretized in the
+          // reference. We remove dimensions that underwent a concretization as
+          // well as the dimensions to the left of that.
+          continue;
+        }
+        break;
```

Comment:
Why break here?

Reply:
I think this part tries to find the domain next to the outer domain, excluding broadcasts. That domain must be the inner domain to keep propagating the contig domains.

Reply:
oh yeah, never mind. I was thinking about something stupid.